Adopt safer CPP in MediaPlayer.cpp

jelee53 opened the pull request:
#### 82787c2128146b297515567d2b7f798289bf4a2a
<pre>
Adopt safer CPP in MediaPlayer.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=297996">https://bugs.webkit.org/show_bug.cgi?id=297996</a>
<a href="https://rdar.apple.com/159317354">rdar://159317354</a>

Reviewed by Rupin Mittal.

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a>

* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::MediaPlayer::~MediaPlayer):
(WebCore::MediaPlayer::prepareForRendering):
(WebCore::MediaPlayer::prepareToPlay):
(WebCore::MediaPlayer::setBufferingPolicy):
(WebCore::MediaPlayer::createSession):
(WebCore::MediaPlayer::setCDM):
(WebCore::MediaPlayer::setCDMSession):
(WebCore::MediaPlayer::keyAdded):
(WebCore::MediaPlayer::cdmInstanceAttached):
(WebCore::MediaPlayer::cdmInstanceDetached):
(WebCore::MediaPlayer::attemptToDecryptWithInstance):
(WebCore::MediaPlayer::setShouldContinueAfterKeyNeeded):
(WebCore::MediaPlayer::setCurrentTimeDidChangeCallback):
(WebCore::MediaPlayer::willSeekToTarget):
(WebCore::MediaPlayer::supportsFullscreen const):
(WebCore::MediaPlayer::canSaveMediaData const):
(WebCore::MediaPlayer::supportsScanning const):
(WebCore::MediaPlayer::supportsProgressMonitoring const):
(WebCore::MediaPlayer::requiresImmediateCompositing const):
(WebCore::MediaPlayer::platformLayer const):
(WebCore::MediaPlayer::updateVideoFullscreenInlineImage):
(WebCore::MediaPlayer::setVideoFullscreenFrame):
(WebCore::MediaPlayer::setVideoFullscreenGravity):
(WebCore::MediaPlayer::setVideoFullscreenMode):
(WebCore::MediaPlayer::videoFullscreenStandbyChanged):
(WebCore::MediaPlayer::setSceneIdentifier):
(WebCore::MediaPlayer::setVideoLayerSizeFenced):
(WebCore::MediaPlayer::timedMetadata const):
(WebCore::MediaPlayer::accessLog const):
(WebCore::MediaPlayer::errorLog const):
(WebCore::MediaPlayer::setVolumeLocked):
(WebCore::MediaPlayer::setVolume):
(WebCore::MediaPlayer::setMuted):
(WebCore::MediaPlayer::hasClosedCaptions const):
(WebCore::MediaPlayer::setClosedCaptionsVisible):
(WebCore::MediaPlayer::rate const):
(WebCore::MediaPlayer::setRate):
(WebCore::MediaPlayer::effectiveRate const):
(WebCore::MediaPlayer::setPreservesPitch):
(WebCore::MediaPlayer::setPitchCorrectionAlgorithm):
(WebCore::MediaPlayer::seekableTimeRangesLastModifiedTime):
(WebCore::MediaPlayer::liveUpdateInterval):
(WebCore::MediaPlayer::setPresentationSize):
(WebCore::MediaPlayer::setVisibleInViewport):
(WebCore::MediaPlayer::setResourceOwner):
(WebCore::MediaPlayer::setPreload):
(WebCore::MediaPlayer::videoFrameForCurrentTime):
(WebCore::MediaPlayer::nativeImageForCurrentTime):
(WebCore::MediaPlayer::shouldGetNativeImageForCanvasDrawing const):
(WebCore::MediaPlayer::supportsPictureInPicture const):
(WebCore::MediaPlayer::isCurrentPlaybackTargetWireless const):
(WebCore::MediaPlayer::wirelessPlaybackTargetType const):
(WebCore::MediaPlayer::wirelessVideoPlaybackDisabled const):
(WebCore::MediaPlayer::setWirelessVideoPlaybackDisabled):
(WebCore::MediaPlayer::canPlayToWirelessPlaybackTarget const):
(WebCore::MediaPlayer::setWirelessPlaybackTarget):
(WebCore::MediaPlayer::setShouldPlayToPlaybackTarget):
(WebCore::MediaPlayer::maxFastForwardRate const):
(WebCore::MediaPlayer::minFastReverseRate const):
(WebCore::MediaPlayer::acceleratedRenderingStateChanged):
(WebCore::MediaPlayer::supportsAcceleratedRendering const):
(WebCore::MediaPlayer::setShouldMaintainAspectRatio):
(WebCore::MediaPlayer::hostingContext const):
(WebCore::MediaPlayer::didPassCORSAccessCheck const):
(WebCore::MediaPlayer::movieLoadType const):
(WebCore::MediaPlayer::mediaTimeForTimeValue const):
(WebCore::MediaPlayer::decodedFrameCount const):
(WebCore::MediaPlayer::droppedFrameCount const):
(WebCore::MediaPlayer::audioDecodedByteCount const):
(WebCore::MediaPlayer::videoDecodedByteCount const):
(WebCore::MediaPlayer::setPrivateBrowsingMode):
(WebCore::MediaPlayer::volumeChanged):
(WebCore::MediaPlayer::audioSourceProvider):
(WebCore::MediaPlayer::waitingForKey const):
(WebCore::MediaPlayer::platformErrorCode const):
(WebCore::MediaPlayer::setTextTrackRepresentation):
(WebCore::MediaPlayer::syncTextTrackBounds):
(WebCore::MediaPlayer::tracksChanged):
(WebCore::MediaPlayer::notifyTrackModeChanged):
(WebCore::MediaPlayer::simulateAudioInterruption):
(WebCore::MediaPlayer::beginSimulatedHDCPError):
(WebCore::MediaPlayer::endSimulatedHDCPError):
(WebCore::MediaPlayer::fileSize const):
(WebCore::MediaPlayer::ended const):
(WebCore::MediaPlayer::setShouldDisableSleep):
(WebCore::MediaPlayer::applicationWillResignActive):
(WebCore::MediaPlayer::applicationDidBecomeActive):
(WebCore::MediaPlayer::objCAVFoundationAVPlayer const):
(WebCore::MediaPlayer::performTaskAtTime):
(WebCore::MediaPlayer::shouldIgnoreIntrinsicSize):
(WebCore::MediaPlayer::isLoopingChanged):
(WebCore::MediaPlayer::setPreferredDynamicRangeMode):
(WebCore::MediaPlayer::setPlatformDynamicRangeLimit):
(WebCore::MediaPlayer::audioOutputDeviceChanged):
(WebCore::MediaPlayer::startVideoFrameMetadataGathering):
(WebCore::MediaPlayer::stopVideoFrameMetadataGathering):
(WebCore::MediaPlayer::renderVideoWillBeDestroyed):
(WebCore::MediaPlayer::setShouldDisableHDR):
(WebCore::MediaPlayer::playerContentBoxRectChanged):
(WebCore::MediaPlayer::supportsPlayAtHostTime const):
(WebCore::MediaPlayer::supportsPauseAtHostTime const):
(WebCore::MediaPlayer::playAtHostTime):
(WebCore::MediaPlayer::pauseAtHostTime):
(WebCore::MediaPlayer::setShouldCheckHardwareSupport):
(WebCore::MediaPlayer::setDefaultSpatialTrackingLabel):
(WebCore::MediaPlayer::setSpatialTrackingLabel):
(WebCore::MediaPlayer::setPrefersSpatialAudioExperience):
(WebCore::MediaPlayer::soundStageSizeDidChange):
(WebCore::MediaPlayer::setInFullscreenOrPictureInPicture):
(WebCore::MediaPlayer::supportsLinearMediaPlayer const):
(WebCore::MediaPlayer::setMessageClientForTesting):
* Source/WebCore/platform/graphics/cocoa/MediaPlayerCocoa.mm:
(WebCore::MediaPlayer::setVideoTarget):

Canonical link: <a href="https://commits.webkit.org/299296@main">https://commits.webkit.org/299296@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d38bdcd7bc7ca4f29a19c1fe8c832dfd850613bb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118534 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38215 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28866 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124706 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70593 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/26c5d589-182a-45c3-aae5-be9a3ba1b1c7) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120412 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38911 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46797 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89969 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59528 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9dc2b0f7-f15c-4ef4-87cb-85f2b31cdd84) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121487 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30993 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106273 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70473 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9f77ea44-3d26-4164-8e25-9b2a43a014d5) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/30050 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24384 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68365 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100432 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24575 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/127772 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45441 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/34285 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/127772 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45805 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102494 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/127772 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/43826 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/21814 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/41939 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18887 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45311 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50989 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/44774 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/48121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46461 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

RupinMittal commented on Source/WebCore/platform/graphics/MediaPlayer.cpp:
```diff
@@ -1442,7 +1442,7 @@ void MediaPlayer::setPrivateBrowsingMode(bool privateBrowsingMode)
     m_inPrivateBrowsingMode = privateBrowsingMode;
     if (m_private)
-        m_private->setPrivateBrowsingMode(m_inPrivateBrowsingMode);
+        protectedPrivate()->setPrivateBrowsingMode(m_inPrivateBrowsingMode);
```
Could do:

```
if (RefPtr private = m_private)
    private->setPrivateBrowsingMode(m_inPrivateBrowsingMode);
````

jelee53 replied:
Oh so I did try that initially. It doesn't work because you can't name a variable "private" because it's a keyword reserved specifically for a qualifier (e.g a private variable). 

I was getting an error similar to this: 
<img width="1163" height="350" alt="Screenshot 2025-08-28 at 3 26 14 PM" src="https://github.com/user-attachments/assets/eb7fabc3-11b2-4a43-935f-77ee801f2be5" />

RupinMittal replied:
Ah of course! Maybe we can call it `RefPtr privateInterface = m_private` instead then (after the type's name `MediaPlayerPrivateInterface`).

RupinMittal commented on Source/WebCore/platform/graphics/MediaPlayer.cpp:
```diff
@@ -1649,23 +1649,23 @@ void MediaPlayer::removeVideoTrack(VideoTrackPrivate& track)
 {
     if (m_private)
-        m_private->notifyTrackModeChanged();
+        protectedPrivate()->notifyTrackModeChanged();
```
Same here

RupinMittal commented on Source/WebCore/platform/graphics/MediaPlayer.cpp:
```diff
@@ -1704,13 +1704,13 @@ bool MediaPlayer::isGStreamerHolePunchingEnabled()
 {
     if (m_private)
-        m_private->beginSimulatedHDCPError();
+        protectedPrivate()->beginSimulatedHDCPError();
```
Same here

RupinMittal commented on Source/WebCore/platform/graphics/MediaPlayer.cpp:
```diff
@@ -1704,13 +1704,13 @@ bool MediaPlayer::isGStreamerHolePunchingEnabled()
 {
     if (m_private)
-        m_private->endSimulatedHDCPError();
+        protectedPrivate()->endSimulatedHDCPError();
```
Same here

RupinMittal commented on Source/WebCore/platform/graphics/MediaPlayer.cpp:
```diff
@@ -1796,7 +1796,7 @@ void MediaPlayer::getRawCookies(const URL& url, MediaPlayerClient::GetRawCookies
 {
     if (m_private)
-        m_private->setShouldDisableSleep(flag);
+        protectedPrivate()->setShouldDisableSleep(flag);
```
Same here

RupinMittal commented on Source/WebCore/platform/graphics/MediaPlayer.cpp:
```diff
@@ -505,8 +505,8 @@ MediaPlayer::MediaPlayer(MediaPlayerClient& client, MediaPlayerEnums::MediaEngin
-    if (m_private)
-        m_private->mediaPlayerWillBeDestroyed();
+    if (RefPtr privatePtr = m_private)
+        privatePtr->mediaPlayerWillBeDestroyed();
```
I think this is good enough:

```
if (RefPtr private = m_private)
    private->mediaPlayerWillBeDestroyed();
```

RupinMittal replied:
Maybe `privateInterface` would be better since we can't use just `private`.

RupinMittal reviewed (approved):
LGTM! Commented some nits.

jelee53 committed:
Adopt safer CPP in MediaPlayer.cpp
https://bugs.webkit.org/show_bug.cgi?id=297996
rdar://159317354

Reviewed by Rupin Mittal.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::MediaPlayer::~MediaPlayer):
(WebCore::MediaPlayer::prepareForRendering):
(WebCore::MediaPlayer::prepareToPlay):
(WebCore::MediaPlayer::setBufferingPolicy):
(WebCore::MediaPlayer::createSession):
(WebCore::MediaPlayer::setCDM):
(WebCore::MediaPlayer::setCDMSession):
(WebCore::MediaPlayer::keyAdded):
(WebCore::MediaPlayer::cdmInstanceAttached):
(WebCore::MediaPlayer::cdmInstanceDetached):
(WebCore::MediaPlayer::attemptToDecryptWithInstance):
(WebCore::MediaPlayer::setShouldContinueAfterKeyNeeded):
(WebCore::MediaPlayer::setCurrentTimeDidChangeCallback):
(WebCore::MediaPlayer::willSeekToTarget):
(WebCore::MediaPlayer::supportsFullscreen const):
(WebCore::MediaPlayer::canSaveMediaData const):
(WebCore::MediaPlayer::supportsScanning const):
(WebCore::MediaPlayer::supportsProgressMonitoring const):
(WebCore::MediaPlayer::requiresImmediateCompositing const):
(WebCore::MediaPlayer::platformLayer const):
(WebCore::MediaPlayer::updateVideoFullscreenInlineImage):
(WebCore::MediaPlayer::setVideoFullscreenFrame):
(WebCore::MediaPlayer::setVideoFullscreenGravity):
(WebCore::MediaPlayer::setVideoFullscreenMode):
(WebCore::MediaPlayer::videoFullscreenStandbyChanged):
(WebCore::MediaPlayer::setSceneIdentifier):
(WebCore::MediaPlayer::setVideoLayerSizeFenced):
(WebCore::MediaPlayer::timedMetadata const):
(WebCore::MediaPlayer::accessLog const):
(WebCore::MediaPlayer::errorLog const):
(WebCore::MediaPlayer::setVolumeLocked):
(WebCore::MediaPlayer::setVolume):
(WebCore::MediaPlayer::setMuted):
(WebCore::MediaPlayer::hasClosedCaptions const):
(WebCore::MediaPlayer::setClosedCaptionsVisible):
(WebCore::MediaPlayer::rate const):
(WebCore::MediaPlayer::setRate):
(WebCore::MediaPlayer::effectiveRate const):
(WebCore::MediaPlayer::setPreservesPitch):
(WebCore::MediaPlayer::setPitchCorrectionAlgorithm):
(WebCore::MediaPlayer::seekableTimeRangesLastModifiedTime):
(WebCore::MediaPlayer::liveUpdateInterval):
(WebCore::MediaPlayer::setPresentationSize):
(WebCore::MediaPlayer::setVisibleInViewport):
(WebCore::MediaPlayer::setResourceOwner):
(WebCore::MediaPlayer::setPreload):
(WebCore::MediaPlayer::videoFrameForCurrentTime):
(WebCore::MediaPlayer::nativeImageForCurrentTime):
(WebCore::MediaPlayer::shouldGetNativeImageForCanvasDrawing const):
(WebCore::MediaPlayer::supportsPictureInPicture const):
(WebCore::MediaPlayer::isCurrentPlaybackTargetWireless const):
(WebCore::MediaPlayer::wirelessPlaybackTargetType const):
(WebCore::MediaPlayer::wirelessVideoPlaybackDisabled const):
(WebCore::MediaPlayer::setWirelessVideoPlaybackDisabled):
(WebCore::MediaPlayer::canPlayToWirelessPlaybackTarget const):
(WebCore::MediaPlayer::setWirelessPlaybackTarget):
(WebCore::MediaPlayer::setShouldPlayToPlaybackTarget):
(WebCore::MediaPlayer::maxFastForwardRate const):
(WebCore::MediaPlayer::minFastReverseRate const):
(WebCore::MediaPlayer::acceleratedRenderingStateChanged):
(WebCore::MediaPlayer::supportsAcceleratedRendering const):
(WebCore::MediaPlayer::setShouldMaintainAspectRatio):
(WebCore::MediaPlayer::hostingContext const):
(WebCore::MediaPlayer::didPassCORSAccessCheck const):
(WebCore::MediaPlayer::movieLoadType const):
(WebCore::MediaPlayer::mediaTimeForTimeValue const):
(WebCore::MediaPlayer::decodedFrameCount const):
(WebCore::MediaPlayer::droppedFrameCount const):
(WebCore::MediaPlayer::audioDecodedByteCount const):
(WebCore::MediaPlayer::videoDecodedByteCount const):
(WebCore::MediaPlayer::setPrivateBrowsingMode):
(WebCore::MediaPlayer::volumeChanged):
(WebCore::MediaPlayer::audioSourceProvider):
(WebCore::MediaPlayer::waitingForKey const):
(WebCore::MediaPlayer::platformErrorCode const):
(WebCore::MediaPlayer::setTextTrackRepresentation):
(WebCore::MediaPlayer::syncTextTrackBounds):
(WebCore::MediaPlayer::tracksChanged):
(WebCore::MediaPlayer::notifyTrackModeChanged):
(WebCore::MediaPlayer::simulateAudioInterruption):
(WebCore::MediaPlayer::beginSimulatedHDCPError):
(WebCore::MediaPlayer::endSimulatedHDCPError):
(WebCore::MediaPlayer::fileSize const):
(WebCore::MediaPlayer::ended const):
(WebCore::MediaPlayer::setShouldDisableSleep):
(WebCore::MediaPlayer::applicationWillResignActive):
(WebCore::MediaPlayer::applicationDidBecomeActive):
(WebCore::MediaPlayer::objCAVFoundationAVPlayer const):
(WebCore::MediaPlayer::performTaskAtTime):
(WebCore::MediaPlayer::shouldIgnoreIntrinsicSize):
(WebCore::MediaPlayer::isLoopingChanged):
(WebCore::MediaPlayer::setPreferredDynamicRangeMode):
(WebCore::MediaPlayer::setPlatformDynamicRangeLimit):
(WebCore::MediaPlayer::audioOutputDeviceChanged):
(WebCore::MediaPlayer::startVideoFrameMetadataGathering):
(WebCore::MediaPlayer::stopVideoFrameMetadataGathering):
(WebCore::MediaPlayer::renderVideoWillBeDestroyed):
(WebCore::MediaPlayer::setShouldDisableHDR):
(WebCore::MediaPlayer::playerContentBoxRectChanged):
(WebCore::MediaPlayer::supportsPlayAtHostTime const):
(WebCore::MediaPlayer::supportsPauseAtHostTime const):
(WebCore::MediaPlayer::playAtHostTime):
(WebCore::MediaPlayer::pauseAtHostTime):
(WebCore::MediaPlayer::setShouldCheckHardwareSupport):
(WebCore::MediaPlayer::setDefaultSpatialTrackingLabel):
(WebCore::MediaPlayer::setSpatialTrackingLabel):
(WebCore::MediaPlayer::setPrefersSpatialAudioExperience):
(WebCore::MediaPlayer::soundStageSizeDidChange):
(WebCore::MediaPlayer::setInFullscreenOrPictureInPicture):
(WebCore::MediaPlayer::supportsLinearMediaPlayer const):
(WebCore::MediaPlayer::setMessageClientForTesting):
* Source/WebCore/platform/graphics/cocoa/MediaPlayerCocoa.mm:
(WebCore::MediaPlayer::setVideoTarget):

Canonical link: https://commits.webkit.org/299296@main
